src/tests/ftest/tags.py

```diff
@@ -650,84 +726,206 @@ def print_verbose(*args):
+        tags (dict): test tags
 
     Returns:
-        set: tags converted to a set
+        str: test frequency
+    """
+    if "pr" in tags:
+        return "pr"
+    elif "daily_regression" in tags:
+        return "daily"
+    elif "weekly_regression" in tags:
+        return "weekly"
+    return "manual"
+
+
+def run_data(paths=None, tags=None, keys=None, csv_file=None):
+    """Display the tests matching the tags and their requested test yaml data keys.
+
+    Args:
+        paths (list, optional): paths to files from which to list via their tags. Defaults to all
+            ftest python files.
+        list (set, optional): list of sets of tags used to filter displayed tests. Defaults to no
+            filtering.
+        list (set, optional): list of sets of test yaml data keys to display. Defaults to None,
+            only displaying the tests.
+        csv_file (str, optional): output file which if specified is generated in a CSV format
+            instead of displaying the test files an their data.
+    """
+    if not paths:
+        paths = all_python_files(FTEST_DIR)
+
+    key_list = sorted(set(chain(*(keys or []))))
+    output = [["Frequency", "File", "Class", "Method"] + key_list]
+
+    tag_map = FtestTagMap(paths)
+    for file_path, classes in iter(tag_map):
+        short_file_path = re.findall(r'ftest/(.*$)', file_path)[0]
+        for class_name, functions in classes.items():
+            for method_name, method_tags in functions.items():
+                if tags and not tag_map.is_test_subset([method_tags], tags):
+                    continue
+
+                # Add a new row of output
+                try:
+                    output.append(
+                        [test_frequency(method_tags), short_file_path, class_name, method_name])
+                    yaml_data = TestYamlData(file_path, method_name)
+                    for key in key_list:
+                        output[-1].append(yaml_data.value(key))
+                except Exception as error:      # pylint: disable=broad-except
+                    print(f"<< Error processing yaml keys {key_list} for {short_file_path} >>")
+                    raise error
+
+    if csv_file:
+        with open(csv_file, 'w', newline='\n') as csvfile:
+            csv_writer = csv.writer(csvfile)
+            for line in output:
+                csv_writer.writerow(line)
+        print(f"Generated {csv_file} with {len(output)} records")
+
+    else:
+        width = []
+        for line in output:
+            for index, column in enumerate(line):
+                size = len(str(column))
+                if index == len(width):
+                    width.append(size)
+                elif size > width[index]:
+                    width[index] = size
+        format_line = "  ".join(list(map(lambda x: f"{{:{x}}}", width)))
+        for line in output:
+            print(format_line.format(*list(map(lambda x: str(x), line))))
```

Comment:
pylint is complaining because lambda isn't needed
```suggestion
            print(format_line.format(*list(map(str, line))))
```

Reply:
Done.